### PR TITLE
[yyjson] Add new port

### PR DESCRIPTION
**Pull request by myd7349**
**Describe the pull request**

- What does your PR fix? Add new port: https://github.com/ibireme/yyjson

- Which triplets are supported/not supported? Have you updated the CI baseline?
N/A. No.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes.

### CODE REVIEW

File: versions/y-/yyjson.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    {
+      "git-tree": "caccd6fff1d6394c7539c43c98079b30b8441e93",
+      "version-date": "2021-04-12",
+      "port-version": 0
+    },
+    {
+      "git-tree": "551a45f891bc48c6f2ba48da1010106a1cca4fbc",
+      "version-date": "2021-04-10",
+      "port-version": 0
+    }
```
Comment (Hoikas) [inline versions/y-/yyjson.json:12]:
```suggestion
    }
```

Comment (myd7349) [reply versions/y-/yyjson.json]: That sounds reasonable. Thanks.

File: ports/yyjson/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    SHA512 05e9cf3e5db5e79188f0207f865744ef411ffef50824783e1d9f4be7ad49fc67aa7490723c44f38c5de336840bbeb948c160bdef9312b97119a6733ff3c1ae04
+    HEAD_REF master
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+	OPTIONS
+	    -DYYJSON_BUILD_TESTS=OFF
+	    -DYYJSON_BUILD_MISC=OFF
```
Comment (NancyLi1013) [inline ports/yyjson/portfile.cmake:14]:
```suggestion
    PREFER_NINJA
    OPTIONS
        -DYYJSON_BUILD_TESTS=OFF
        -DYYJSON_BUILD_MISC=OFF
```

Comment (NancyLi1013) [reply ports/yyjson/portfile.cmake:14]: This function has been deprecated in favor of `vcpkg_cmake_configure` from the `vcpkg-cmake` port.

So could you please update it? See the reference https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_configure_cmake.md

File: ports/yyjson/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+vcpkg_install_cmake()
```
Comment (NancyLi1013) [inline ports/yyjson/portfile.cmake:17]: The same as above.

File: ports/yyjson/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+vcpkg_copy_pdbs()
+
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/${PORT})
```
Comment (NancyLi1013) [inline ports/yyjson/portfile.cmake:21]: The same as above.

File: ports/yyjson/vcpkg.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "yyjson",
+  "version-date": "2021-04-12",
+  "description": "A high performance JSON library written in ANSI C",
+  "homepage": "https://github.com/ibireme/yyjson",
+  "license": "MIT"
+}
```
Comment (NancyLi1013) [inline ports/yyjson/vcpkg.json:7]: If we use the new functions, `vcpkg-cmake` and `vcpkg-cmake-config` are required to be added as dependencies.

```
"dependencies": [
    {
      "name": "vcpkg-cmake",
      "host": true
    },
    {
      "name": "vcpkg-cmake-config",
      "host": true
    }
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/yyjson/portfile.cmake (hunk 2 above)
Comment (myd7349) [reply ports/yyjson/portfile.cmake]: @NancyLi1013 Thanks! Didn't realize the existance of these new functions.

File: ports/yyjson/portfile.cmake (hunk 3 above)
Comment (myd7349) [reply ports/yyjson/portfile.cmake]: Thanks.

File: ports/yyjson/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO ibireme/yyjson
+    REF 31313e8c15e3c221c7452fd99bf7fdf89b6d92c1
+    SHA512 05e9cf3e5db5e79188f0207f865744ef411ffef50824783e1d9f4be7ad49fc67aa7490723c44f38c5de336840bbeb948c160bdef9312b97119a6733ff3c1ae04
+    HEAD_REF master
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
```
Comment (NancyLi1013) [inline ports/yyjson/portfile.cmake:11]:
```suggestion
```
This argument seems not needed here.

By default, when possible, vcpkg_cmake_configure uses ninja-build as its build system. If the WINDOWS_USE_MSBUILD argument is passed, then vcpkg_cmake_configure will use a Visual Studio generator on Windows; on every other platform, vcpkg_cmake_configure just uses Ninja.
https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/ports/vcpkg-cmake/vcpkg_cmake_configure.md

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/yyjson/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    OPTIONS
+        -DYYJSON_BUILD_TESTS=OFF
+        -DYYJSON_BUILD_MISC=OFF
+)
+
+vcpkg_cmake_install()
+
+vcpkg_copy_pdbs()
+
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/${PORT})
```
Comment (NancyLi1013) [inline ports/yyjson/portfile.cmake:20]: Is the `PACKAGE_NAME ` `yyjson ` here?

Comment (myd7349) [reply ports/yyjson/portfile.cmake:20]: Yes.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks for adding this new port @myd7349.

**Comment by myd7349** [comment]
Hi! @NancyLi1013 Thank you for your kind help.

**Comment by BillyONeal** [comment]
Thanks for the new port :D

**Pull request merged**